app/src/main/java/openfoodfacts/github/scrachx/openfood/models/Product.java

```diff
@@ -162,231 +162,126 @@ public void setAdditionalProperty(String name, Object value) {
     }
 
     public String getProductName(String languageCode) {
-        if (!languageCode.equals("en") && additionalProperties.get("product_name_" + languageCode) != null
-                && isNotBlank(additionalProperties.get("product_name_" + languageCode).toString())) {
-            return additionalProperties.get("product_name_" + languageCode)
-                    .toString()
-                    .replace("\\'", "'")
-                    .replace("&quot", "'");
-        } else if (additionalProperties.get("product_name_en") != null
-                && isNotBlank(additionalProperties.get("product_name_en").toString())) {
-            return additionalProperties.get("product_name_en")
-                    .toString()
-                    .replace("\\'", "'")
-                    .replace("&quot", "'");
+        if (getFieldHelper("product_name", languageCode) != null) {
```

Comment:
One last thing, you should probably save the result of `getFieldHelper` as a local variable so you don't have to fetch it twice. Other than that, everything else looks good.